lighthouse-core/lib/url-shim.js

```diff
@@ -215,6 +215,32 @@ class URLShim extends URL {
     const urlScheme = protocol.includes(':') ? protocol.slice(0, protocol.indexOf(':')) : protocol;
     return NON_NETWORK_SCHEMES.includes(urlScheme);
   }
+
+  /**
+   * @param {string} src
+   * @return {string|undefined}
+   */
+  static guessMimeType(src) {
+    let url;
+    try {
+      url = new URL(src);
+    } catch {
+      return undefined;
+    }
+
+    if (url.protocol === 'data:') {
+      const match = url.pathname.match(/image\/(png|jpeg|jpg|svg\+xml|webp|gif|avif);/);
```

Comment:
just alternative to avoid the `split`, lookaheads can be confusing though, so understood if you'd rather keep the regex simpler

```suggestion
      const match = url.pathname.match(/image\/(png|jpeg|jpg|svg\+xml|webp|gif|avif)(?=;)/);
```